clients/include/testing_syr2_batched.hpp

```diff
@@ -0,0 +1,255 @@
+    double rocblas_gflops, cblas_gflops, rocblas_bandwidth;
+    double rocblas_error_1;
+    double rocblas_error_2;
+
+    // Initial Data on CPU
+    rocblas_init(hA_1, true);
+    rocblas_init(hx, false);
+    rocblas_init(hy, false);
+
+    hA_2.copy_from(hA_1);
+    hA_gold.copy_from(hA_1);
+
+    CHECK_HIP_ERROR(dA_1.transfer_from(hA_1));
+    CHECK_HIP_ERROR(dA_2.transfer_from(hA_1));
+    CHECK_HIP_ERROR(dx.transfer_from(hx));
+    CHECK_HIP_ERROR(dy.transfer_from(hy));
+
+    if(arg.unit_check || arg.norm_check)
+    {
+        // copy data from CPU to device
+        //CHECK_HIP_ERROR(hipMemcpy(dA_2, hA_2, sizeof(T) * lda * N, hipMemcpyHostToDevice));
+        CHECK_HIP_ERROR(d_alpha.transfer_from(halpha));
+
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+        CHECK_ROCBLAS_ERROR(rocblas_syr2_batched<T>(handle,
+                                                    uplo,
+                                                    N,
+                                                    &h_alpha,
+                                                    dx.ptr_on_device(),
+                                                    incx,
+                                                    dy.ptr_on_device(),
+                                                    incy,
+                                                    dA_1.ptr_on_device(),
+                                                    lda,
+                                                    batch_count));
+
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_device));
+        CHECK_ROCBLAS_ERROR(rocblas_syr2_batched<T>(handle,
+                                                    uplo,
+                                                    N,
+                                                    d_alpha,
+                                                    dx.ptr_on_device(),
+                                                    incx,
+                                                    dy.ptr_on_device(),
+                                                    incy,
+                                                    dA_2.ptr_on_device(),
+                                                    lda,
+                                                    batch_count));
+
+        CHECK_HIP_ERROR(hA_1.transfer_from(dA_1));
+        CHECK_HIP_ERROR(hA_2.transfer_from(dA_2));
+
+        // CPU BLAS
+        cpu_time_used = get_time_us();
+        for(int i = 0; i < batch_count; i++)
+        {
+            cblas_syr2<T>(uplo, N, h_alpha, hx[i], incx, hy[i], incy, hA_gold[i], lda);
+        }
+        cpu_time_used = get_time_us() - cpu_time_used;
+        cblas_gflops  = batch_count * syr2_gflop_count<T>(N) / cpu_time_used * 1e6;
+
+        if(arg.unit_check)
+        {
+            unit_check_general<T>(1, size_A, batch_count, 1, hA_gold, hA_1);
+            unit_check_general<T>(1, size_A, batch_count, 1, hA_gold, hA_2);
```

Comment:
Don't these fail without near check for complex?

Reply:
Hm, I changed this to unit_checks for all syr2 tests and they still all seem to pass. I'll fiddle around and see what happens with syr/spr/spr2/etc.

Reply:
They all seem to be passing with unit_checks. I'm not sure which functions in particular were originally failing with unit_checks. I can change the near_checks in all these to unit_checks in another PR later and see how they do on CI.

Reply:
Sounds fine, it seems suspicious if they accumulate complex values I would have thought they would need near.

Reply:
> near_check

Like `unit_check_general`, `near_check_general` now has two template arguments, which should be the same for the majority of functions.